lib/perimeterx/internal/validators/perimeter_x_cookie_validator.rb

```diff
@@ -24,18 +28,23 @@ def verify(px_ctx)
             @logger.warn("PerimeterxCookieValidator:[verify]: Empty token value - decryption failed")
             px_ctx.context[:s2s_call_reason] = PxModule::COOKIE_DECRYPTION_FAILED
             return false, px_ctx
-          elsif px_ctx.context[:px_cookie] == "1"
-            @logger.warn("PerimeterxCookieValidator:[verify]: no cookie")
-            px_ctx.context[:s2s_call_reason] = PxModule::NO_COOKIE
-            return false, px_ctx
-          elsif px_ctx.context[:px_cookie] == "2"   # Mobile SDK connection error
-            @logger.warn("PerimeterxCookieValidator:[verify]: mobile sdk connection error")
-            px_ctx.context[:s2s_call_reason] = PxModule::MOBILE_SDK_CONNECTION_ERROR
-            return false, px_ctx
-          elsif px_ctx.context[:px_cookie] == "3"   # Mobile SDK pinning error
-            @logger.warn("PerimeterxCookieValidator:[verify]: mobile sdk pinning error")
-            px_ctx.context[:s2s_call_reason] = PxModule::MOBILE_SDK_PINNING_ERROR
-            return false, px_ctx
+          elsif px_ctx.context[:px_cookie].to_s =~ /^\d{1,4}$/
+            @logger.warn("PerimeterxCookieValidator:[verify]: Mobile SDK Error Received")
+            px_ctx.context[:s2s_call_reason] = PxModule::MOBILE_SDK_ERROR_MESSAGE + px_ctx.context[:px_cookie]
+            if px_ctx.context[:headers][PxModule::ORIGINAL_TOKEN_HEADER]
+              token = force_utf8(px_ctx.context[:headers][PxModule::ORIGINAL_TOKEN_HEADER])
+              @logger.warn("the contents of token are #{token}")
+              if token.include? ':'
+                exploded_token = token.split(':', 2)
+                cookie_sym = "v#{exploded_token[0]}".to_sym
+                @logger.warn("I made it to here and token version is #{cookie_sym}")
+                px_ctx.context[:px_cookie][cookie_sym] = exploded_token[1]
```

Comment:
@chennakarpx this is the line of code that is causing me to receive the error:

`E, [2020-08-25T14:05:36.082825 #1] ERROR -- : PerimeterxCookieValidator:[verify]: exception while verifying cookie => no implicit conversion of Symbol into Integer`